File: Week_03/G20200343040030/n-ary-tree-postorder-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function(root) {
  let res = [];
  function helper(root) {
    if (!root) return;
    let children = root.children;
    for (let i = 0; i < children.length; i++) {
      helper(children[i]);
    }
    res.push(root.val);
  }
  helper(root);
  return res;
};

File: Week_03/G20200343040044/nAryTreePostorderTraversal.java
package postorderTraversal;

import java.util.ArrayList;
import java.util.List;

/**
 * @Author VITI.lewis
 * @Despriction:
 * @Date:Created in 2020/3/29 9:16 下午
 * @Modify By:
 */
public class nAryTreePostorderTraversal {

    public List<Integer> postorder(Node root) {
        List<Integer> list = new ArrayList<Integer>();
        if (root == null) {
            return list;
        }
        for (Node node:root.children) {
            list.addAll(postorder(node));
        }
        list.add(root.val);
        return  list;
    }

}
